### PR TITLE
Use correct pos-only arg formatting in `pretty_callable`

### LINKED ISSUE #13346: (🐞) Positional only parameter syntax doesn't show in overload error messages
```py
from typing import overload

@overload
def foo(a: int, /): ...
@overload
def foo(a: str): ...
def foo(a): ...


foo(a=1)
# main.py:10: error: No overload variant of "foo" matches argument type "int"
# main.py:10: note: Possible overload variants:
# main.py:10: note:     def foo(a, int) -> Any
# main.py:10: note:     def foo(a: str) -> Any
```
The error message is now incorrect.

[playground](https://mypy-play.net/?mypy=latest&python=3.10&gist=dfafdca4f2bc9e23a4b004ffeea72f80)

### PR DESCRIPTION
**Pull request by sobolevn**
This is right now a WIP, because I want to take a look at what will happen with tests, mypy-primer, etc.

Later I will add new tests for new format and consider all corner cases: like pos-only `self` or `cls` parameters.

Closes #13346

### CODE REVIEW

**Comment by sobolevn** [comment]
This one looks wrong: 

```
discord/http.py:226: note:     def __setitem__(self, slice, /, Iterable[Embed], /) -> None
```

**Review by JelleZijlstra — APPROVED** [review]

File: mypy/messages.py [hunk 1]
```diff
@@ -2394,23 +2395,29 @@ def [T <: int] f(self, x: int, y: T) -> None
...
     ):
-        definition_args = [arg.variable.name for arg in tp.definition.arguments]
-        if (
-            definition_args
-            and tp.arg_names != definition_args
-            and len(definition_args) > 0
-            and definition_args[0]
-        ):
+        definition_arg_names = [arg.variable.name for arg in tp.definition.arguments]
+        if len(definition_arg_names) > len(tp.arg_names) and definition_arg_names[0]:
```
Comment (hauntsaninja) [inline mypy/messages.py:2417]: Btw is it true that `len(definition_arg_names) == len(tp.arg_names) or len(definition_arg_names) == len(tp.arg_names) + 1`? i.e. is there ever some case with multiple special args?

**Review by hauntsaninja — APPROVED** [review]
Looks great, thanks for improving this!

File: mypy/messages.py (hunk 1 above)
Comment (sobolevn) [reply mypy/messages.py:2417]: I don't think there are any such cases.

**Pull request merged**